### PR TITLE
Check input profile name for duplicates in New/Rename

### LINKED ISSUE #4611: Controller profiles do no name duplication checking
It is possible to create several named profiles with the same name. This obviously isn't intended here and will confuse users. Fixing this should be easy by just checking if a profile with the same name already exists.

Screenshot:
unbenannt

cc @valentinvanelslande

### PR DESCRIPTION
**Pull request by Mates1500**
Closes #4611

Perhaps I could shove the condition into a separate function instead of copypasting it? I would certainly do that in C#, but I'm not particularly familiar with C++ codebases and coding styles, so let me know.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4779)

### CODE REVIEW

**Comment by wwylele** [comment]
LGTM in general. Putting the condition + message box into a separate function is indeed preferably here to avoid duplicating the translation string.

Travis CI somehow didn't start. You might want to check clang-format manually (or let travis CI check it on the next push)

**Comment by Mates1500** [comment]
> Edit: Name suggestion for helper function (since naming can be annoying sometimes haha): `bool IsProfileNameDuplicate` and have it popup the message box, and return true if theres no issues/false if theres a duplicate. That way the calling functions can still early return in the error case.

Done
I'm used to long descriptive class and function names from the C# world. Though `IsProfileNameDuplicate()` would not implicate the MessageBox behavior, so I've separated that into its own function as well.

File: src/citra_qt/configuration/configure_input.cpp [hunk 1]
```diff
@@ -465,6 +470,23 @@ void ConfigureInput::RenameProfile() {
...
+        WarnProposedProfileNameIsDuplicate();
+        return;
+    }
+
     ui->profile->setItemText(ui->profile->currentIndex(), new_name);
     Settings::RenameCurrentProfile(new_name.toStdString());
 }
+
+bool ConfigureInput::IsProfileNameDuplicate(const QString& name) {
+    if (ui->profile->findText(name, Qt::MatchFixedString | Qt::MatchCaseSensitive) != -1) {
```
Comment (jroweboy) [inline src/citra_qt/configuration/configure_input.cpp]: super super nitpick:

```cpp
bool ConfigureInput::IsProfileNameDuplicate(const QString& name) {
    return ui->profile->findText(name, Qt::MatchFixedString | Qt::MatchCaseSensitive) != -1;
}
```

Comment (Mates1500) [reply src/citra_qt/configuration/configure_input.cpp]: that's more of a personal preference, but sure

Comment (jroweboy) [reply src/citra_qt/configuration/configure_input.cpp]: Oh, i wasn't trying to imply it was wrong, just trying to keep the style of the project consistent <3 Thanks a bunch for the contribution again! I'll merge this when CI passes

File: src/citra_qt/configuration/configure_input.h [hunk 2]
```diff
@@ -119,4 +119,7 @@ public slots:
     void NewProfile();
     void DeleteProfile();
     void RenameProfile();
+
+    bool IsProfileNameDuplicate(const QString& name);
```
Comment (lioncash) [inline src/citra_qt/configuration/configure_input.h]:
```
bool IsProfileNameDuplicate(const QString& name) const;
```

**Pull request merged**